### PR TITLE
[DX-1235] Return expected response types

### PR DESCRIPTION
**Pull request by matthewmuscat**
# Summary
This PR returns the expected API response interface types to remove the existing `AxiosResponse` wrapper. It also handles errors cleaner as a source of truth for consumers. It also regenerates the spec.

Main commits for CR reference:
- https://github.com/immutable/imx-core-sdk/pull/172/commits/e3f655c0281b106e9b672b435a91fdf6358ae2b0
- https://github.com/immutable/imx-core-sdk/pull/172/commits/f8d5ab3721f1aa1584986529b8207d400a25cbe4
- https://github.com/immutable/imx-core-sdk/pull/172/commits/934ade253d24c97c2b4d884cecdcaec0c5550c02

# Why the changes
To improve overall DX for consumers when handling errors, and for public method return types to not include the `AxiosPromise` wrapper anymore.

Example flow:
```ts
import { ImmutableX, IMXError } from '@imtbl/core-sdk';

try {
  const client = new ImmutableX(config);
  // mock throws
  const balance = await client.getBalance({
    owner: await walletConnection.ethSigner.getAddress(),
    address: userEthAddress,
  });
} catch (err: unknown) {
  if (err instanceof IMXError) {
    // handle appropriately

    console.log(err)
    // {
    //   code: 'resource_not_found_code',
    //   message: "Token id '0xd6d44dd0270d46a710e04deacb787617ee262459' not found",
    //   details: undefined,
    // }
  }
}
```

# Things worth calling out

### CODE REVIEW

File: src/api/domain/metadata-refreshes-api.ts [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+    readonly createMetadataRefreshRequest: CreateMetadataRefreshRequest
+}
+
+/**
+ * MetadataRefreshesApi - object-oriented interface
+ * @export
+ * @class MetadataRefreshesApi
+ * @extends {BaseAPI}
+ */
+export class MetadataRefreshesApi extends BaseAPI {
```
Comment (zatlite) [inline src/api/domain/metadata-refreshes-api.ts:585]: Not a blocker for this PR but we will need these methods from the new MetadataRefreshesApi to ImmutableX class.

**Review by zatlite — APPROVED** [review]
Looks great!

**Review by CodeSchwert — APPROVED** [review]

File: src/api/domain/metadata-refreshes-api.ts (hunk 1 above)
Comment (Kastet) [reply src/api/domain/metadata-refreshes-api.ts:585]: Yup. Can be done separately or even post 1.0.

File: src/ImmutableX.ts [hunk 2]
```diff
@@ -96,116 +97,186 @@ export class ImmutableX {
    * @param ethSigner - the L1 signer
    * @param deposit - the token type amount in its corresponding unit
    * @returns the response of the resulting transaction
+   * @throws IMXError
    */
   public deposit(ethSigner: EthSigner, deposit: TokenAmount) {
-    return this.workflows.deposit(ethSigner, deposit);
+    return this.workflows
+      .deposit(ethSigner, deposit)
+      .then(res => res)
```
Comment (Kastet) [inline src/ImmutableX.ts:105]: Does `.then(res => res)` do anything? Seems unnecessary.

Comment (Kastet) [reply src/ImmutableX.ts:105]: If not required remove from other workflows too

File: src/ImmutableX.ts [hunk 3]
```diff
@@ -316,90 +411,137 @@ export class ImmutableX {
...
+      .catch(err => {
+        throw formatError(err);
+      });
   }
 
   /**
    * Get the token Balances of the User
    * @param request - the request object containing the parameters to be provided in the API request
-   * @returns the requested Balance
+   * @returns a promise that resolves with the requested Balance
```
Comment (Kastet) [inline src/ImmutableX.ts:425]: this return type description is now inconsistent with other methods, probably worth reverting or changing everywhere

File: src/types/errors.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        message: String(err),
+      });
+    }
+  } else {
+    return new IMXError({
+      code: UnknownErrorCode,
+      message: String(err),
+    });
+  }
+}
```
Comment (Kastet) [inline src/types/errors.ts:52]: can be simplified 
```
export function formatError(err: unknown): IMXError {
  if ((axios.isAxiosError(err) && err.response) {
    const error: APIError = err.response.data;
    return new IMXError({
      code: error.code,
      details: error.details,
      message: error.message,
    });
  }
  return new IMXError({
    code: UnknownErrorCode,
    message: String(err),
  });
}
```

File: src/ImmutableX.ts (hunk 2 above)
Comment (matthewmuscat) [reply src/ImmutableX.ts]: Not needed 👍

File: src/types/errors.ts (hunk 4 above)
Comment (matthewmuscat) [reply src/types/errors.ts]: nice! addressed https://github.com/immutable/imx-core-sdk/pull/172/commits/958ecc16a2bd2af4c7fd2bfd9defda71736e63d2

File: src/ImmutableX.ts (hunk 3 above)
Comment (matthewmuscat) [reply src/ImmutableX.ts:416]: I prefer the prefixed promise mention, it adds better context. have updated all other methods.

Comment (matthewmuscat) [reply src/ImmutableX.ts:416]: https://github.com/immutable/imx-core-sdk/pull/172/commits/958ecc16a2bd2af4c7fd2bfd9defda71736e63d2

File: src/ImmutableX.ts (hunk 2 above)
Comment (matthewmuscat) [reply src/ImmutableX.ts]: https://github.com/immutable/imx-core-sdk/pull/172/commits/958ecc16a2bd2af4c7fd2bfd9defda71736e63d2

**Review by Kastet — APPROVED** [review]
👌

**Pull request merged**